contracts/controller/src/admin.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(dead_code)]
+
+use near_sdk::{AccountId, env};
+
+use crate::Contract;
+
+pub enum MetricType {
+    ReserveFactor,
+    HealthFactor,
+    LiquidationIncentive,
+}
+
+pub enum MethodType {
+    Withdraw,
+    Repay,
+    Supply,
+    Liquidate,
+    Borrow,
+}
+
+
+impl Contract {
+    fn get_admin(&self) -> AccountId {
+        return self.admin.clone();
+    }
+
+    fn set_admin(&mut self, account: AccountId) {
+        assert!(self.is_valid_admin_call(), "This functionality is allowed to be called by admin or contract only");
+        self.admin = account;
+    }
+
+    pub fn is_valid_admin_call(&self) -> bool {
+        env::predecessor_account_id() == self.admin || env::predecessor_account_id() == env::current_account_id()
+    }
+
+
+    fn add_market_asset(&mut self, key: AccountId, value: AccountId) {
+        assert!(self.is_valid_admin_call(), "This functionality is allowed to be called by admin or contract only");
+
+        self.markets.insert(&key, &value);
+    }
+
+    fn remove_market_asset(&mut self, key: AccountId) {
+        assert!(self.is_valid_admin_call(), "This functionality is allowed to be called by admin or contract only");
+
+        assert!(self.markets.contains_key(&key), "Asset by this key doesnt exist");
+
+        self.markets.remove(&key);
+    }
+
+
+    fn get_metric(self, user: AccountId, metric: MetricType) -> u128 {
```

Comment:
@dmytro-shash you don't need a unified getter for all those fields because they are not represent state of a single field. In this case separate getters are not violation of DRY, fruthermore current implementation is violation of Single responsibility principle (from SOLID) because getter for Reserve factor shouldn't know anything about HealthFactor and LiquidatorIncentive